integrations/datadog/queries.yaml

```diff
@@ -0,0 +1,184 @@
+    - replica_name:
+        usage: "LABEL"
+        description: "Name of this cluster replica"
+    - cluster_type:
+        usage: "LABEL"
+        description: "Name of this cluster"
+    - cpu_percent:
+        usage: "GAUGE"
+        description: "CPU percentage usage of this cluster."
+    - memory_percent:
+        usage: "GAUGE"
+        description: "Memory percentage usage of this cluster."
+    - credits_per_hour:
+        usage: "GAUGE"
+        description: "Credits usage of this cluster."
+sink_statistics:
+  query: |
+          SELECT
+            S.name as sink_name,
+            S.type as sink_type,
+            S.cluster_id,
+            C.name as cluster_name,
+            CASE status
+              WHEN 'created' THEN 0
+              WHEN 'starting' THEN 1
+              WHEN 'running' THEN 2
+              WHEN 'stalled' THEN 3
+              WHEN 'failed' THEN 4
+              WHEN 'dropped' THEN 5
+              ELSE 6
+            END AS sink_status,
+            SUM(messages_committed) as messages_committed,
+            SUM(bytes_committed) as bytes_committed
+          FROM mz_internal.mz_sink_statistics SS
+          JOIN mz_catalog.mz_sinks S ON (SS.id = S.id)
+          JOIN mz_catalog.mz_clusters C ON (S.cluster_id = C.id)
+          JOIN mz_internal.mz_source_statuses ST ON (S.id = ST.id)
+          GROUP BY sink_name, sink_type, cluster_id, cluster_name, sink_status;
+  master: true
+  metrics:
+    - sink_name:
+        usage: "LABEL"
+        description: "Name of this sink"
+    - sink_type:
+        usage: "LABEL"
+        description: "Type of this sink"
+    - cluster_id:
+        usage: "LABEL"
+        description: "Identifier of this sink's cluster"
+    - cluster_name:
+        usage: "LABEL"
+        description: "Name of this sink's cluster"
+    - sink_status:
+        usage: "GAUGE"
+        description: "Status of this sink"
+    - messages_committed:
+        usage: "COUNTER"
+        description: "Messages commited by this sink."
+    - bytes_committed:
+        usage: "COUNTER"
+        description: "Bytes commited by this sink."
+source_statistics:
+  query: |
+          SELECT
+            S.name as source_name,
+            S.type as source_type,
+            S.cluster_id,
+            C.name as cluster_name,
+            CASE status
+              WHEN 'created' THEN 0
+              WHEN 'starting' THEN 1
+              WHEN 'running' THEN 2
+              WHEN 'stalled' THEN 3
+              WHEN 'failed' THEN 4
+              WHEN 'dropped' THEN 5
+              ELSE 6
+            END AS source_status,
+            bool_and(snapshot_committed)::int::float as snapshot_committed,
+            SUM(SS.envelope_state_bytes) as envelope_state_bytes,
+            SUM(SS.envelope_state_count) as envelope_state_count,
+            SUM(messages_received) as messages_received,
+            SUM(bytes_received) as bytes_received
+          FROM mz_internal.mz_source_statistics SS
```

Comment:
By leading with `mz_source_statistics`/`mz_sink_statistics`, we can potentially end up with no data for these metrics (e.g. load generator sources). Should the lead table be `mz_source_statuses`/`mz_sink_statuses` instead?